src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -131,8 +197,62 @@ public Uri GetProxy(Uri uri)
         {
             if (_proxyHelper.ManualSettingsOnly)
             {
+                if (_bypassLocal)
+                {
+                    IPAddress address = null;
+
+                    if (uri.IsLoopback)
+                    {
+                        // This is optimization for loopback addresses.
+                        // Unfortunately this does not work for all local addresses.
+                        return null;
+                    }
+
+                    // Pre-Check if host may be IP address to avoid parsing.
+                    if (uri.Host[0] == '[' || (uri.Host[0] >= '0' && uri.Host[0] <= '9'))
```

Comment:
uri.Host is not particularly cheap.  It'd be better to access it once and store it into a local.